// wasm-fractals/wasm_fractals_bg.js
import * as wasm from './wasm_fractals_bg.wasm';

const lTextDecoder = typeof TextDecoder === 'undefined' ? (0, module.require)('util').TextDecoder : TextDecoder;

let cachedTextDecoder = new lTextDecoder('utf-8', { ignoreBOM: true, fatal: true });

cachedTextDecoder.decode();

let cachedUint8Memory0 = new Uint8Array();

function getUint8Memory0() {
    if (cachedUint8Memory0.byteLength === 0) {
        cachedUint8Memory0 = new Uint8Array(wasm.memory.buffer);
    }
    return cachedUint8Memory0;
}

function getStringFromWasm0(ptr, len) {
    return cachedTextDecoder.decode(getUint8Memory0().subarray(ptr, ptr + len));
}
/**
*/
export class JuliaSet {

    static __wrap(ptr) {
        const obj = Object.create(JuliaSet.prototype);
        obj.ptr = ptr;

        return obj;
    }

    __destroy_into_raw() {
        const ptr = this.ptr;
        this.ptr = 0;

        return ptr;
    }

    free() {
        const ptr = this.__destroy_into_raw();
        wasm.__wbg_juliaset_free(ptr);
    }
    /**
    * @param {number} iterations
    * @param {number} cx
    * @param {number} cy
    * @returns {JuliaSet}
    */
    static new(iterations, cx, cy) {
        const ret = wasm.juliaset_new(iterations, cx, cy);
        return JuliaSet.__wrap(ret);
    }
    /**
    */
    render() {
        wasm.juliaset_render(this.ptr);
    }
    /**
    * @returns {number}
    */
    width() {
        const ret = wasm.juliaset_width(this.ptr);
        return ret >>> 0;
    }
    /**
    * @returns {number}
    */
    height() {
        const ret = wasm.juliaset_height(this.ptr);
        return ret >>> 0;
    }
    /**
    * @returns {number}
    */
    hues() {
        const ret = wasm.juliaset_hues(this.ptr);
        return ret;
    }
}

export function __wbindgen_throw(arg0, arg1) {
    throw new Error(getStringFromWasm0(arg0, arg1));
};
